File: javascript/leetcode/Perform String Shifts.js
/**
 * @param {string} s
 * @param {number[][]} shift
 * @return {string}
 */
var stringShift = function (s, shift) {
  const str = [];
  shift.forEach((i) => {
    let amount = i[1];

    if (i[0] === 0) {
      while (amount--) {
        s += s[0];
        s = s.slice(1);
      }
    } else {
      while (amount--) {
        let end = s[s.length - 1];
        s = end + s.slice(0, s.length - 1);
      }
    }
  });
  return s;
};

File: c++/leetcode/Perform String Shifts.cpp
class Solution
{
public:
  string stringShift(string s, vector<vector<int>> &shift)
  {
    for (auto i : shift)
    {

      auto amount = i[1];

      if (i[0] == 0)
      {
        while (amount--)
        {
          s.push_back(s.front());
          s.erase(s.begin());
        }
      }
      else
      {
        while (amount--)
        {
          s.insert(s.begin(), s.back());
          s.pop_back();
        }
      }
    }

    return s;
  }
};